pallets/reward-distribution/src/lib.rs

```diff
@@ -182,22 +183,44 @@ pub mod pallet {
 				return Err(Error::<T>::NoRewardsForAccount.into())
 			}
 
-			//either transfer from fee account if the reward is
-			//not the native currency, or just mint it if it is
-			let native_currency_id = T::GetNativeCurrencyId::get();
-			if reward_currency_id == native_currency_id {
-				T::Balances::deposit_creating(&caller, rewards);
+			//transfer rewards
+			Self::transfer_reward(reward_currency_id, reward, caller)
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	fn transfer_reward(
+		reward_currency_id: CurrencyId<T>,
+		reward: BalanceOf<T>,
+		beneficiary: T::AccountId,
+	) -> DispatchResult {
+		let native_currency_id = T::GetNativeCurrencyId::get();
+		if reward_currency_id == native_currency_id {
+			let available_native_funds = T::Balances::total_balance(&Self::fee_pool_account_id());
+
+			if available_native_funds < reward {
+				//if available native is less than reward we transfer first from fee account,
+				//and mint the rest
+				let remaining =
+					reward.checked_sub(&available_native_funds).ok_or(Error::<T>::Underflow)?;
+				let amount: currency::Amount<T> =
+					Amount::new(available_native_funds, reward_currency_id);
+				amount.transfer(&Self::fee_pool_account_id(), &beneficiary)?;
+				T::Balances::deposit_creating(&beneficiary, remaining);
 				return Ok(())
 			} else {
-				let amount: currency::Amount<T> = Amount::new(rewards, reward_currency_id);
-				amount.transfer(&Self::fee_pool_account_id(), &caller)?;
-				return Ok(())
+				let amount: currency::Amount<T> = Amount::new(reward, reward_currency_id);
+				return amount.transfer(&Self::fee_pool_account_id(), &beneficiary)
 			}
+		} else {
+			//we need no checking of available funds, since fee will ALWAYS have enough collected
+			//fees
```

Comment:
```suggestion
			//we need no checking of available funds, since the fee pool will ALWAYS have enough collected
			//fees of the wrapped currencies

```